### PR TITLE
Codespaces

### PR DESCRIPTION
**Pull request by peteryates**
This change enables Codespaces for the ECF app. This will help less technical colleagues make direct changes.

The only change that has any risk of affecting the app is the one to the initializer that lets us easily specify the `REDIS_URI` using a environment variable instead of letting it default to `localhost`.

Note, in its current form this PR adds `entr` as a dependency for running `bin/dev` - not ideal but I can't find a way of making Middleman continuously monitor for changes (like `middleman serve`) but actually output built files (like middleman build`). @DFE-Digital/teacher-cpd - any ideas?

| Preview |
| ------- |
| image |

### CODE REVIEW

**Review by cwrw — APPROVED** [review]
🙌

File: config/initializers/sidekiq.rb [hunk 1]
```diff
@@ -23,6 +23,16 @@
   end
 end
 
+if ENV.key?("REDIS_URI")
```
Comment (peteryates) [inline config/initializers/sidekiq.rb:26]: `REDIS_URI` is set in the GitHub Codespaces environment variable settings. As per the the docker compose it's `redis://redis`.

**Comment by peteryates** [comment]
So it turns out building the guide using Foreman in a Codespace that uses rvm doesn't play nicely.

I've removed the `api_docs` line from the Procfile and added a section to the readme detailing how to run it separately. I left `entr` in because it should work fine on a mac and is easily installable with brew/macports - but if it turns out to be problematic for any reason happy to switch to another.

**Review by cwrw — APPROVED** [review]
tested new approach this morning and works nicely, thanks!

**Comment by peteryates** [comment]
@slorek - mind doing a rereview? Requests have been addressed. Ta.

**Pull request merged**